main.go

```diff
@@ -13,6 +13,15 @@ import (
 
 func main() {
 	log.SetFormatter(&log.TextFormatter{ForceColors: true})
+	log.SetOutput(colorable.NewColorableStderr())
+	msg := `
+aks-engine is deprecated for Azure public cloud customers.
+Please consider using Azure Kubernetes Service (AKS) for managed Kubernetes:
+    https://azure.microsoft.com/en-us/services/kubernetes-service/#overview
+or Cluster API Provider Azure (CAPZ) for self-managed Kubernetes:
+    https://capz.sigs.k8s.io/
+`
+	log.Warningf("\u001b[33m%s\u001b[0m", msg)
```

Comment:
Can we do one of these stdout UT?

https://github.com/Azure/aks-engine/blob/master/cmd/get_locations_test.go#L68

Or does the color + formatting subvert that?

Mind you, unit-testing stdout is not highest priority, and including accessibility-friendly colored output is much more valuable!

Reply:
I looked hard at writing a unit test, but the simplest place to put this code so it's always called is in the `main` func, which isn't covered by unit tests.

Maybe we could change the implementation so it was in each cobra Command, so we could test it there. (I also looked at `Command.Deprecated` and `Command.Flags().MarkDeprecated()` and neither quite does what we want here.)

Reply:
> unit-testing stdout is not highest priority

I think there's not enough value in unit testing this particular code to justify rewriting it.